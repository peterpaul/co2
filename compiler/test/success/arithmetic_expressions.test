["stdio.h"] int printf(char * format, ...);

int main(int argc, char ** argv)
{
	int i = 5;
	printf("i (5) = %d\n", i);
	printf("i+1 (6) = %d\n", i+1);
	printf("i*2 (10) = %d\n", i*2);
	printf("i-2 (3) = %d\n", i-2);
	printf("i/5 (1) = %d\n", i/5);
	printf("i^3 (125) = %d\n", i^3);
	printf("i%3 (2) = %d\n", i%3);
	printf("i+=1 (6) = %d\n", i+=1);
	printf("i*=2 (12) = %d\n", i*=2);
	printf("i-=2 (10) = %d\n", i-=2);
	printf("i/=5 (2) = %d\n", i/=5);
	printf("i^=3 (8) = %d\n", i^=3);
	printf("i%=3 (2) = %d\n", i%=3);

	unsigned u = 5;
	printf("u (5) = %u\n", u);
	printf("u+1 (6) = %u\n", u+1);
	printf("u*2 (10) = %u\n", u*2);
	printf("u-2 (3) = %u\n", u-2);
	printf("u/5 (1) = %u\n", u/5);
	printf("u^3 (125) = %u\n", u^3);
	printf("u%3 (2) = %u\n", u%3);
	printf("u+=1 (6) = %u\n", u+=1);
	printf("u*=2 (12) = %u\n", u*=2);
	printf("u-=2 (10) = %u\n", u-=2);
	printf("u/=5 (2) = %u\n", u/=5);
	printf("u^=3 (8) = %u\n", u^=3);
	printf("u%=3 (2) = %u\n", u%=3);

	float f = 5.0;
	printf("f (5) = %f\n", f);
	printf("f+1 (6) = %f\n", f+1.0);
	printf("f*2 (10) = %f\n", f*2.0);
	printf("f-2 (3) = %f\n", f-2.0);
	printf("f/5 (1) = %f\n", f/5.0);
	printf("f^3 (125) = %f\n", f^3.0);
	printf("f%3 (2) = %f\n", f%3.0);
	printf("f+=1 (6) = %f\n", f+=1.0);
	printf("f*=2 (12) = %f\n", f*=2.0);
	printf("f-=2 (10) = %f\n", f-=2.0);
	printf("f/=5 (2) = %f\n", f/=5.0);
	printf("f^=3 (8) = %f\n", f^=3.0);
	printf("f%=3 (2) = %f\n", f%=3.0);
	return 0;
}