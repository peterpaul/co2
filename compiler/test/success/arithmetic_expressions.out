i (5) = 5
i+1 (6) = 6
i*2 (10) = 10
i-2 (3) = 3
i/5 (1) = 1
i^3 (125) = 125
i%3 (2) = 2
i+=1 (6) = 6
i*=2 (12) = 12
i-=2 (10) = 10
i/=5 (2) = 2
i^=3 (8) = 8
i%=3 (2) = 2
u (5) = 5
u+1 (6) = 6
u*2 (10) = 10
u-2 (3) = 3
u/5 (1) = 1
u^3 (125) = 125
u%3 (2) = 2
u+=1 (6) = 6
u*=2 (12) = 12
u-=2 (10) = 10
u/=5 (2) = 2
u^=3 (8) = 8
u%=3 (2) = 2
f (5) = 5.000000
f+1 (6) = 6.000000
f*2 (10) = 10.000000
f-2 (3) = 3.000000
f/5 (1) = 1.000000
f^3 (125) = 125.000000
f%3 (2) = 2.000000
f+=1 (6) = 6.000000
f*=2 (12) = 12.000000
f-=2 (10) = 10.000000
f/=5 (2) = 2.000000
f^=3 (8) = 8.000000
f%=3 (2) = 2.000000
